### ISSUE TITLE
[Request] Deleted instances should go to the recycling bin

### ISSUE THREAD

**Issue by aaronhowser1** [MultiMC/MultiMC5#3193]
System Information
-----------------------------
MultiMC version: 0.6.11-1430

Operating System: W10

Summary of the issue or suggestion: When an instance is deleted, it should be sent to the recycling bin so it can be recovered
----------------------------------------------

**Comment by Heufneutje** [comment]
Why? MultiMC asks you very clearly if you're sure you want to delete your instance. Aside from that it would have to be implemented separately for each OS. And not even all of them have recycle bins.

**Comment by aaronhowser1** [comment]
Because some people ~~are stupid~~ might delete the wrong instance if they aren't paying attention, and it would prevent [something like this](https://www.distractify.com/p/deleted-sons-minecraft-world-aita)

**Comment by Heufneutje** [comment]
It just adds an unnecessary extra step for everyone who does want to actually delete an instance, not to mention maintainability cost. All to solve a very niche use case that most games don't bother with either. If you delete a save file it's gone as well. I don't see this any differently.

**Comment by Zetabite** [comment]
I agree to @Heufneutje its just an extra hassle and I have nobody than myself to blame if I delete an instance. Yes it can happen by accident, but thats still my fault.

**Comment by peterix** [comment]
That's what backups are for.

But if I were to do this, I'd implement a 'recycle-bin-like' functionality in MultiMC, not use the system one. They are incredibly janky and unmanageable from within MultiMC.

**Comment by JakeMoe** [comment]
If this does get implemented (which I personally hope it doesn't), I hope it'd be a) a configurable option, and b) off by default.  I certainly don't want this; I hard-delete everything in Windows, bypassing the Recycle Bin, so it doesn't clutter up my disk.  In Linux, I mostly use the command line to delete, so it doesn't apply.  And with modpacks (and all the mods inside them) taking up as much space as they do, that MultiMC recycle bin could get pretty big.

**Comment by ProjectSynchro** [comment]
Something like this could easily be an opt in feature, But seeing that an 'ARE YOU SURE' prompt comes up when you opt to delete an instance..

I don't really see why it would be needed.

**Comment by codingJWilliams** [comment]
The kind of people who opt in to that feature are not the same people that randomly delete instances they need without taking a backup. I don't see the point.

**Comment by catronomix** [comment]
Today I accidentally removed my main instance instead of a single screenshot because I thought I activated a windows Explorer window but I unknowingly focused the MultiMC window. Lost weeks of progress (years if I had not made a copy weeks ago which luckily I did). But still I was surprised that pressing delete and not shift+delete would skip the recycle bin!

Of course its a form of user error, but that's the whole reason for OS's to implement things like the recycle bin.
Saying not every OS has that feature is a moot argument. It can easily be accounted for by calling the OS for the folder deletion operation, and from there on the operation will be performed in a way the user is familiar with.

Disk space issues that would arise by keeping the deleted instances should also be handled by the OS's cleanup features when applicable.

Bottom line is, making it super easy to delete someone's years upon years of creative progress without any chance to undo that action when it is technically possible to give it a chance is not the right choice when it's just to make a point along the lines of: "if you make an error it's your own fault".

**Comment by Zetabite** [comment]
If someone comes around implementing it in a way that makes sense and makes a PR, sure. But as of now, its as "wont fix"/"out of scope".

**Comment by phit** [comment]
dont put your faith into the recycle bin to keep anything you care about safe, drives die all the time or get destroyed, local and external backups!

**Comment by Heufneutje** [comment]
> Disk space issues that would arise by keeping the deleted instances should also be handled by the OS's cleanup features when applicable.

How exactly would the OS know to specifically clean up that folder? As stated before, if this were to be implemented it wouldn't use the system recycle bin.

**Comment by catronomix** [comment]
I'm not really a programmer, just some very basic knowledge. But from what I find out from searching this repo, the GUI is using this QT framework, for which the documentation lists this function: `QFile::moveToTrash()` which I guess is supposed to use the trash when the OS supports it?

**Comment by malino24** [comment]
I've just had a case where I deleted an instance without knowing the worlds are gone too. Should have known since I looked at the folder structure before but guess I just didn't. But the worst part is that I seemingly can't even use file recovery software to get my backup (which MMC has created because of a version switch) back. So I don't know what's going on here that even this software can't find the *.zip file. But to actually contribute to this issue: a possible solution would be to at least have backupped worlds to not be deleted when deleting an instance.
